### PR TITLE
Feature Add getWithTags method

### PR DESCRIPTION
**Pull request by cameronsmith**
Added getWithTags method.

### CODE REVIEW

**Comment by kibertoad** [comment]
@cameronsmith Thanks! Can you add documentation for it?

File: lib/awilixManager.ts [hunk 1]
```diff
@@ -8,15 +8,21 @@ declare module 'awilix' {
     asyncDispose?: boolean | string | ((instance: T) => Promise<unknown>)
     asyncDisposePriority?: number // lower means it gets disposed earlier
     eagerInject?: boolean | string
+    tags?: string[]
     enabled?: boolean
   }
 }
 
+interface AssociativeArray {
```
Comment (kibertoad) [inline lib/awilixManager.ts]: That's just a `Record<string, object>`

File: lib/awilixManager.ts [hunk 2]
```diff
@@ -8,15 +8,21 @@ declare module 'awilix' {
...
+  [key: string]: object
+}
+
 export type AwilixManagerConfig = {
   diContainer: AwilixContainer
   asyncInit?: boolean
   asyncDispose?: boolean
   eagerInject?: boolean
+  tags?: string[]
```
Comment (kibertoad) [inline lib/awilixManager.ts]: what is this flag used for?

File: lib/awilixManager.ts [hunk 3]
```diff
@@ -83,6 +89,21 @@ export function eagerInject(diContainer: AwilixContainer) {
   }
 }
 
+export function getWithTags(diContainer: AwilixContainer, tags: string[]): AssociativeArray {
```
Comment (kibertoad) [inline lib/awilixManager.ts]: I would suggest typing it as `Record<string, any>`

File: test/awilixManager.spec.ts [hunk 4]
```diff
@@ -107,6 +108,30 @@ describe('awilixManager', () => {
       expect(dependency3.isInitted).toBe(true)
     })
 
+    it('execute asyncInit on registered dependencies with valid tags', async () => {
+      const diContainer = createContainer({
+        injectionMode: 'PROXY',
+      })
+      diContainer.register(
```
Comment (kibertoad) [inline test/awilixManager.spec.ts:114]: maybe add more dependencies to check that only the right ones are included?

File: lib/awilixManager.ts (hunk 2 above)
Comment (cameronsmith) [reply lib/awilixManager.ts]: This is the option where the user can define the tags for their dependency, e.g.

```
diContainer.register(
        'dependency2',
        asClass(AsyncInitGetClass, {
          lifetime: 'SINGLETON',
          asyncInit: true,
          asyncInitPriority: 1,
          tags: ['engine', 'ChatGPT-4'] <<< HERE
        }),
      )
```

Without this we get a type error.

Comment (kibertoad) [reply lib/awilixManager.ts]: there are two configs, one for manager, one for dependency. this one is for manager and I don't think we use it

File: README.md [hunk 5]
```diff
@@ -100,3 +100,48 @@ const awilixManager = new AwilixManager({
...
+## Fetching dependencies based on tags
+
+In some cases you may want to get dependencies based on a supplied list of tags. 
+You can use `tags` parameter in conjunction with the `getWithTags` method for that:
+
+```js
+import { AwilixManager, getWithTags } from 'awilix-manager'
+import { asClass, createContainer } from 'awilix'
+
+class QueueConsumerHighPriortyClass {
```
Comment (kibertoad) [inline README.md]:
```suggestion
class QueueConsumerHighPriorityClass {
```

File: README.md [hunk 6]
```diff
@@ -100,3 +100,48 @@ const awilixManager = new AwilixManager({
...
+  
+}
+
+class QueueConsumerLowPriortyClass {
```
Comment (kibertoad) [inline README.md]:
```suggestion
class QueueConsumerLowPriorityClass {
```

File: lib/awilixManager.ts [hunk 7]
```diff
@@ -83,6 +85,21 @@ export function eagerInject(diContainer: AwilixContainer) {
   }
 }
 
+export function getWithTags(diContainer: AwilixContainer, tags: string[]): Record<string, object> {
```
Comment (kibertoad) [inline lib/awilixManager.ts]: `any` would be better than `object` here, as theoretically nothing prevents you from using DI for a boolean or a number

**Review by kibertoad — APPROVED** [review]

**Pull request merged**